File: RepoTransBench/target_projects/Python/C++/LanceGin_haishoku/tests/original/test_haishoku.cpp
#include <gtest/gtest.h>
#include "haishoku/haishoku.h"

TEST(TestHaishoku, test_Haishoku_instance) {
    // In Python: returns class (not an object), so mimic that
    auto obj = Haishoku::loadHaishoku("demo/demo_01.png");
    EXPECT_EQ(obj, typeid(Haishoku));
    // Further tests can be implemented once the function is properly defined.
}

File: RepoTransBench/target_projects/Python/Rust/LanceGin_haishoku/tests/original/test_haishoku.rs
use haishoku_rs::haishoku::Haishoku;

#[test]
fn test_hs_instance() {
    // The test expects load_haishoku to return the class itself.
    let class_ptr = Haishoku::load_haishoku("demo/demo_01.png");
    // We can test that it is actually the static instance, which acts as a "class"
    // or at least, the test must confirm that calling the function gives a constant address.
    let class_ptr2 = Haishoku::load_haishoku("demo/demo_01.png");
    assert_eq!(class_ptr as *const _, class_ptr2 as *const _);
}